File: Player.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Player
{
    class Point
    {
        public int X { get; set; }
        public int Y { get; set; }
    }

    class Drone
    {
        public Point Position { get; set; } 
    }

    class Zone
    {
        public const int Radius = 100;

        public Point Center { get; set; }
        public int OwnerId { get; set; } // -1 if no owner
    }

    class Team
    {
        public Team(int droneCount)
        {
            Drones = new List<Drone>(droneCount);
            for (var droneId = 0; droneId < droneCount; droneId++)
                Drones.Add(new Drone());
        }

        public IList<Drone> Drones { get; private set; }
    }

    class Game
    {
        List<Zone> zones; // all game zones
        List<Team> teams; // all the team of drones. Array index = team's ID
        int myTeamId; // index of my team in the array of teams

        // read initial games data (one time at the beginning of the game: P I D Z...)
        public void Init()
        {
            var pidz = ReadIntegers();

            myTeamId = pidz[1];
            zones = ReadZones(pidz[3]).ToList();
            teams = ReadTeams(pidz[0], pidz[2]).ToList();
        }

        IEnumerable<Zone> ReadZones(int zoneCount)
        {
            for (var areaId = 0; areaId < zoneCount; areaId++)
                yield return new Zone {Center = ReadPoint()};
        }

        IEnumerable<Team> ReadTeams(int teamCount, int dronesPerTeam)
        {
            for (var teamId = 0; teamId < teamCount; teamId++)
                yield return new Team(dronesPerTeam);
        }

        public void ReadContext()
        {
            foreach (var zone in zones)
                zone.OwnerId = ReadIntegers()[0];

            foreach (var team in teams)
                foreach (var drone in team.Drones)
                    drone.Position = ReadPoint();
        }

        // Compute logic here. This method is called for each game round. 
        public void Play()
        {
            var myDrones = teams[myTeamId].Drones;

            // here I always ask my drones to reach the bottom right corner... stupid :-) 
            foreach (var drone in myDrones)
            {
                Console.WriteLine("3999 1799");

            }
        }

        static int[] ReadIntegers()
        {
            // ReSharper disable once PossibleNullReferenceException
            return Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
        }

        static Point ReadPoint()
        {
            var xy = ReadIntegers();
            return new Point { X = xy[0], Y = xy[1] };
        }
    }

    static class Program
    {
        static void Main()
        {
            var game = new Game();

            game.Init();
            
            while (true)
            {
                game.ReadContext();
                game.Play();
            }
        }
    }
}

File: Player.rb
STDOUT.sync = true

class Point
  attr_accessor :x, :y
  def initialize(x, y)
    @x = x
    @y = y
  end
end

class Drone
  attr_accessor :position
  def initialize
    @position = nil
  end
end

class Zone
  attr_accessor :center, :owner_id
  RADIUS = 100
  def initialize(center)
    @center = center
    @owner_id = -1
  end
end

class Team
  attr_accessor :drones
  def initialize(drone_count)
    @drones = []
    (0..drone_count - 1).each do
      @drones.push Drone.new
    end
  end
end

class Game
  def init
    @zones = []
    @teams = []

    @players_count, @my_team_id, @drones_count, @zones_count = read_integers
    read_zones
    read_teams
  end

  # Compute logic here. This method is called for each game round.
  def play
    my_drones = @teams[@my_team_id].drones

    my_drones.each do |drone|
      puts "3999 1799"
    end
  end

  def read_context
    @zones.each do |zone|
      zone.owner_id = read_integers[0]
    end

    @teams.each do |team|
      team.drones.each do |drone|
        drone.position = read_point
      end
    end
  end

  def read_zones
    (0..@zones_count - 1).each do
      @zones.push Zone.new(read_point)
    end
  end

  def read_teams
    (0..@players_count - 1).each do
      @teams.push Team.new(@drones_count)
    end
  end
end

def read_point
  xy = read_integers
  Point.new(xy[0], xy[1])
end

def read_integers
  STDIN.gets.split.map { |s| s.to_i}
end

game = Game.new

game.init

loop do
  game.read_context
  game.play
end

